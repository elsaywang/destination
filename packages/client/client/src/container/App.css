@import '~@spectrum/spectrum-variables/css/spectrum-light.css';
