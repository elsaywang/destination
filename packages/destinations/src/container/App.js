import React, { Fragment } from 'react';
import './App.css';
import { Provider } from 'react-redux';
import { Route, BrowserRouter, Switch, Redirect } from 'react-router-dom';
import Nav from '../components/Nav';
import Layout from '../components/Layout';
import Destinations from './Destinations';
import Configuration from './Configuration';
import configureStore from '../configureStore';
import { routes } from '../constants/navTab';
const store = configureStore();

const App = () => (
    <BrowserRouter basename="/portal">
        <Switch>
            <Route
                exact
                path={'/administration/integrated-accounts'}
                component={IntegratedAccounts}
            />
            <Route exact path="/destinations" render={() => <Redirect to="/destinations/all" />} />
            {routes.map(routeObject => (
                <Route
                    key={routeObject.route}
                    path={routeObject.route}
                    render={() => <DestinationContainer routeObject={routeObject} />}
                />
            ))}
        </Switch>
    </BrowserRouter>
);

const DestinationContainer = ({ routeObject }) => (
    <Provider store={store}>
        <Fragment>
            <Layout heading="Destinations">
                <Nav routes={routes} />
                <Route
                    path={routeObject.route}
                    render={() => <Destinations currentDestination={routeObject} />}
                />
            </Layout>
        </Fragment>
    </Provider>
);

const IntegratedAccounts = () => (
    <Layout heading="Integrated Accounts">
        <Configuration />
    </Layout>
);
export default App;
