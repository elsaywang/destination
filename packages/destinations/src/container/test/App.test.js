import React from 'react';
import { shallow } from 'enzyme';
import App from '../App';

describe('renders <App/> without crashing', () => {
    const wrapper = shallow(<App />);

    describe('rendering', () => {
        it('matches snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        });
    });
});
