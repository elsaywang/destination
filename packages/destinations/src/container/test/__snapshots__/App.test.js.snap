// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <App/> without crashing rendering matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "basename": "/portal",
      "children": <Switch>
        <Route
                component={[Function]}
                exact={true}
                path="/administration/integrated-accounts"
        />
        <Route
                exact={true}
                path="/destinations"
                render={[Function]}
        />
        <Route
                path="/destinations/all"
                render={[Function]}
        />
        <Route
                path="/destinations/adobeExperienceCloud"
                render={[Function]}
        />
        <Route
                path="/destinations/integratedPlatforms"
                render={[Function]}
        />
        <Route
                path="/destinations/custom"
                render={[Function]}
        />
</Switch>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Route
            component={[Function]}
            exact={true}
            path="/administration/integrated-accounts"
/>,
          <Route
            exact={true}
            path="/destinations"
            render={[Function]}
/>,
          Array [
            <Route
              path="/destinations/all"
              render={[Function]}
/>,
            <Route
              path="/destinations/adobeExperienceCloud"
              render={[Function]}
/>,
            <Route
              path="/destinations/integratedPlatforms"
              render={[Function]}
/>,
            <Route
              path="/destinations/custom"
              render={[Function]}
/>,
          ],
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/administration/integrated-accounts",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/destinations",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "/destinations/all",
          "nodeType": "class",
          "props": Object {
            "path": "/destinations/all",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "/destinations/adobeExperienceCloud",
          "nodeType": "class",
          "props": Object {
            "path": "/destinations/adobeExperienceCloud",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "/destinations/integratedPlatforms",
          "nodeType": "class",
          "props": Object {
            "path": "/destinations/integratedPlatforms",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "/destinations/custom",
          "nodeType": "class",
          "props": Object {
            "path": "/destinations/custom",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "basename": "/portal",
        "children": <Switch>
          <Route
                    component={[Function]}
                    exact={true}
                    path="/administration/integrated-accounts"
          />
          <Route
                    exact={true}
                    path="/destinations"
                    render={[Function]}
          />
          <Route
                    path="/destinations/all"
                    render={[Function]}
          />
          <Route
                    path="/destinations/adobeExperienceCloud"
                    render={[Function]}
          />
          <Route
                    path="/destinations/integratedPlatforms"
                    render={[Function]}
          />
          <Route
                    path="/destinations/custom"
                    render={[Function]}
          />
</Switch>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              exact={true}
              path="/administration/integrated-accounts"
/>,
            <Route
              exact={true}
              path="/destinations"
              render={[Function]}
/>,
            Array [
              <Route
                path="/destinations/all"
                render={[Function]}
/>,
              <Route
                path="/destinations/adobeExperienceCloud"
                render={[Function]}
/>,
              <Route
                path="/destinations/integratedPlatforms"
                render={[Function]}
/>,
              <Route
                path="/destinations/custom"
                render={[Function]}
/>,
            ],
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/administration/integrated-accounts",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/destinations",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "/destinations/all",
            "nodeType": "class",
            "props": Object {
              "path": "/destinations/all",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "/destinations/adobeExperienceCloud",
            "nodeType": "class",
            "props": Object {
              "path": "/destinations/adobeExperienceCloud",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "/destinations/integratedPlatforms",
            "nodeType": "class",
            "props": Object {
              "path": "/destinations/integratedPlatforms",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "/destinations/custom",
            "nodeType": "class",
            "props": Object {
              "path": "/destinations/custom",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
